lib/azure/rest.rb

```diff
@@ -58,7 +58,8 @@ def http_query(request_url, verb, body, params)
     end
 
     def query_for_completion()
-      request_url = "https://#{@host_name}/#{@subscription_id}/operations/#{@last_request_id}"
+      uri = URI.parse("#{@host_name}/#{@subscription_id}/operations/#{@last_request_id}")
+      request_url = "#{get_scheme_for_request_url(uri)}#{@host_name}/#{@subscription_id}/operations/#{@last_request_id}"
```

Comment:
Instead of `get_scheme_for_request_url` method, we can have a method `get_request_url`. We should pass the `uri` to it and it should return the complete `url`.